File: source/includes/fts/divide-results/simple-query.java
import java.util.Arrays;
import static com.mongodb.client.model.Projections.excludeId;
import static com.mongodb.client.model.Projections.fields;
import static com.mongodb.client.model.Projections.include;
import static com.mongodb.client.model.Aggregates.limit;
import static com.mongodb.client.model.Aggregates.skip;
import static com.mongodb.client.model.Aggregates.project;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class DivideQueryResults {

    public static void main(String[] args) {
        // connect to your Atlas cluster
        String uri = "<connection-string>";
        
        try (MongoClient mongoClient = MongoClients.create(uri)) {
            // set namespace
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");
            
            // define pipeline
            Document agg = new Document("$search", new Document("index", "pagination-tutorial")
            		.append("text", new Document("query", "tom hanks").append("path", "cast")));
            
            // run pipeline and print results
            collection.aggregate(Arrays.asList(agg,
            	skip(10),
                limit(10),
                project(fields(excludeId(), include("title"), include("cast")))))
            .forEach(doc -> System.out.println(doc.toJson()));
        }
    }
}

File: source/includes/fts/divide-results/simple-query.kt
import com.mongodb.client.model.Aggregates.*
import com.mongodb.client.model.Projections.*
import com.mongodb.kotlin.client.coroutine.MongoClient
import kotlinx.coroutines.runBlocking
import org.bson.Document

fun main() {
    // connect to your Atlas cluster
    val uri = "<connection-string>"
    val mongoClient = MongoClient.create(uri)

    // set namespace
    val database = mongoClient.getDatabase("sample_mflix")
    val collection = database.getCollection<Document>("movies")

    runBlocking {
        // define pipeline
        val agg = Document(
            "\$search", Document("index", "pagination-tutorial")
                .append("text", Document("query", "tom hanks").append("path", "cast"))
        )

        // run pipeline and print results
        val resultsFlow = collection.aggregate<Document>(
            listOf(
                agg,
                skip(10),
                limit(10),
                project(fields(
                    excludeId(),
                    include("title","cast")
                ))
            )
        )
        resultsFlow.collect { println(it) }
    }
    mongoClient.close()
}